crafters/image/ImageTorchTransformation/tests/test_imagetorchtransformation.py

```diff
@@ -0,0 +1,217 @@
+        ([normalize_transform()], org_img(), normalize_img()),
+    ],
+)
+def test_common_transforms(transforms, input, expected):
+    crafter = ImageTorchTransformation(transforms)
+    crafted_imgs = crafter.craft(input)
+    np.testing.assert_almost_equal(expected, crafted_imgs[0]["blob"], decimal=6)
+
+
+@pytest.mark.parametrize(
+    "transforms, input, expected",
+    [
+        ([resize_transform()], org_img(), resize_img()),
+    ],
+)
+def test_transform_visually(transforms, input, expected):
+    # visual inspection only - WIP https://github.com/pytorch/vision/issues/2950
+    crafter = ImageTorchTransformation(transforms)
+    crafted_imgs = crafter.craft(input)
+    cv2.imwrite(
+        f"{cur_dir}/{list(transforms[0].keys())[0].lower()}_out.png",
+        cv2.cvtColor(crafted_imgs[0]["blob"] * 255, cv2.COLOR_RGB2BGR),
+    )
+
+
+@pytest.mark.parametrize(
+    "transforms, inputs, expected",
+    [
+        (
+            [center_crop_transform(), normalize_transform()],
+            [org_img(), org_img()],
+            normalize_image(center_crop_img() * 255),
+        ),
+        (
+            [flip_transform(), normalize_transform()],
+            [org_img(), org_img()],
+            normalize_image(vflip_img() * 255),
+        ),
+    ],
+)
+def test_sequential_transforms(transforms, inputs, expected):
+    crafter = ImageTorchTransformation(transforms)
+    crafted_imgs = crafter.craft(inputs)
+
+    assert len(crafted_imgs) == 2
+    for crafted_img in crafted_imgs:
+        np.testing.assert_almost_equal(expected, crafted_img["blob"], decimal=6)
+
+
+@pytest.mark.parametrize(
+    "transforms, input, expected",
+    [
+        ([{"RandomVerticalFlip": dict(p=0.0)}], org_img(), vflip_img()),
+        ([{"RandomVerticalFlip": dict(p=1.0)}], org_img(), vflip_img()),
+    ],
+)
+def test_random_transforms(transforms, input, expected):
+    # Here randomness is in if or not to do transform. It is overrided to always do.
+    crafter = ImageTorchTransformation(transforms)
+    crafted_imgs = crafter.craft(input)
+    np.testing.assert_almost_equal(expected, crafted_imgs[0]["blob"], decimal=6)
+
+
+def test_incorrect_transforms():
+    # transform should be list
+    with pytest.raises(ValueError):
+        ImageTorchTransformation("RandomVerticalFlip")
+    # incorrect transform name
+    with pytest.raises(ValueError):
+        ImageTorchTransformation(["Fake"])
+    # incorrect arg
+    with pytest.raises(ValueError):
+        ImageTorchTransformation([{"RandomVerticalFlip": dict(Fake=1.0)}])
+    # allowed single image array (H, B, C), list of image arrays, image batch array (B, H, W, C)
+    with pytest.raises(AssertionError):
+        ImageTorchTransformation(["RandomVerticalFlip"]).craft(np.zeros((100, 100)))
+    # transform not supported for tensor
+    with pytest.raises(ValueError):
+        ImageTorchTransformation(["RandomChoice"])
+
+
+def test_example():
```

Comment:
I think this test would be good to be tested in batches as well

Reply:
You mean, `test_incorrect_transforms()`? This is only for testing the initialization of `Compose` alone.
`test_allowed_argument_types` Line 76. tests single and batch inputs for the transforms. By batch, I mean a list of (2) images or numpy array (B, H, W, C) 

I guess I didn't get your suggestion.

Reply:
> Are there tests with inputs in the batches with different shapes?

Yes I have this 
```
@pytest.mark.parametrize(
    "transforms, inputs",
    [
        (["ColorJitter"], org_img()),
        (["ColorJitter"], np.stack([org_img()] * 3)),
        ([{"GaussianBlur": dict(kernel_size=(3, 3))}], [org_img(), org_img()]),
    ],
)
def test_allowed_argument_types(transforms, inputs):
    # All valid formats of transforms and inputs
```

Reply:
what I mean is that u test the results match for an image given as a single input, not in batches